pkg/apis/eksctl.io/v1alpha3/types.go

```diff
@@ -181,31 +227,49 @@ func (c *ClusterConfig) NewNodeGroup() *NodeGroup {
 // NodeGroup holds all configuration attributes that are
 // specific to a nodegroup
 type NodeGroup struct {
-	Name string
-
-	AMI               string
-	AMIFamily         string
-	InstanceType      string
-	AvailabilityZones []string
-	Tags              map[string]string
-	PrivateNetworking bool
-
-	DesiredCapacity int
-	MinSize         int
-	MaxSize         int
-
-	VolumeSize int
-
-	Labels         NodeLabels
-	MaxPodsPerNode int
-
-	PolicyARNs      []string
-	InstanceRoleARN string
-
-	AllowSSH         bool
-	SSHPublicKeyPath string
-	SSHPublicKey     []byte
-	SSHPublicKeyName string
+	Name string `json:"name"`
+	// +optional
+	AMI string `json:"ami,omitempty"`
+	// +optional
+	AMIFamily string `json:"amiFamily,omitempty"`
+	// +optional
+	InstanceType string `json:"instanceType,omitempty"`
+	// +optional
+	AvailabilityZones []string `json:"availabilityZones,omitempty"`
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+	// +optional
+	PrivateNetworking bool `json:"privateNetworking"`
+
+	// +optional
+	DesiredCapacity int `json:"desiredCapacity"`
```

Comment:
Could the "desired" capacity not be met?
Would it be clearer to have something named like, e.g., the below?
```
Count int `json:"count"`
```

Reply:
Yes, we had it as `NodeCount` at the top level some time ago, but now it's called `DesiredCapacity` to match the AWS API. However, the flag is still `--nodes`, so there is a little inconsistency there too. Some refactoring is pending here also (#385).

Reply:
Fair enough 😅 
Matching the AWS API probably takes precedence over any other argument/preference 👍